nginx-controller/controller/status.go

```diff
@@ -67,12 +69,26 @@ func (su *StatusUpdater) updateIngressWithStatus(ing v1beta1.Ingress, status []a
 	if reflect.DeepEqual(ing.Status.LoadBalancer.Ingress, status) {
 		return nil
 	}
-	// Objects from the LoadBalancerController.ingLister.Store are retrieved by reference
-	// and are not safe to modify.
-	ingCopy := ing.DeepCopy()
+
+	// get a pristine ing with no annotation changes
```

Comment:
Please explain why need to do that. For example. `get a pristine ing. required because annotations can be modified for mergeable Ingress resources and the update status API call updates annotations`